mne/viz/raw.py

```diff
@@ -552,13 +552,16 @@ def _setup_channel_selections(raw, kind, order):
     from ..selection import (read_selection, _SELECTIONS, _EEG_SELECTIONS,
                              _divide_to_regions)
     from ..utils import _get_stim_channel
+    from .. import BaseEpochs
     _check_option('group_by', kind, ('position', 'selection'))
     if kind == 'position':
         selections_dict = _divide_to_regions(raw.info)
         keys = _SELECTIONS[1:]  # omit 'Vertex'
     else:  # kind == 'selection'
         from ..io import RawFIF, RawArray
-        if not isinstance(raw, (RawFIF, RawArray)):
+        if isinstance(raw, BaseEpochs):
+            pass
+        elif not isinstance(raw, (RawFIF, RawArray)):
```

Comment:
@larsoner is there a clever way to detect whether an Epochs object contains Neuromag data?  I was never satisfied for Raw checking against RawFIF and RawArray, so if you have a better idea on that too I'd love to hear it.

Reply:
If this has to do with `selection` mode, I assume you are talking about Neuromag MEG data (as EEG can be treated like any other format). The robust way would be to check for coil_type of any of info['chs'] being one of the Neuromag types. Part of this sort of functionality might already live in find_layout somehow, because it needs to inspect channels to see if we have MEG channels that we can use a nice built-in layout for.